src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/DeletionReferralService.java

```diff
@@ -60,7 +60,15 @@ public void handlePendingDeletion(final OffenderPendingDeletionEvent event) {
     }
 
     public void handleReferralComplete(final OffenderPendingDeletionReferralCompleteEvent event) {
-        // TODO GDPR-99 Track referral requests and add health check
+
+        log.info("All offenders pending deletion in batch: '{}' have been added to the queue", event.getBatchId());
+
+        final var batch = batchRepository.findById(event.getBatchId())
+                .orElseThrow(illegalState("Cannot find batch with id: '%s'", event.getBatchId()));
```

Comment:
looks familiar! :-D

Reply:
Haha yep, I do miss some of those nice little utilities :D